app/lib/world/GoalManager.coffee

```diff
@@ -269,6 +312,10 @@ module.exports = class GoalManager extends CocoClass
 
   setGoalState: (goalID, status) ->
     state = @goalStates[goalID]
+    if not state
+      console.error('Failed to set goal state, possibly due to capstone stages for non-ozaria play')
+      return
```

Comment:
Why is this required?

Reply:
During the development of this feature, the additional goals were loaded in immediately upon completing the last additional goal of the stage. This caused challenges because the state was not propagated throughout the entire state of the website. 

These guards were added to avoid other areas of the code trying to look up the state that didn't exist yet. 

As the modals now always redirect to the cinematic placeholder (and the cinematic in the future), this is no longer an issue and these can be cleaned up 👍

Reply:
There are still cases where this part of the code would be hit as part of updates before transitioning out of the play view, where it is useful to have graceful handling instead of erroring because the state can't be found.